pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -713,8 +714,12 @@ func PodFitsResources(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *s
 	}
 
 	var podRequest *schedulercache.Resource
+	scalaResourcesHandleByExtenders := sets.NewString()
 	if predicateMeta, ok := meta.(*predicateMetadata); ok {
 		podRequest = predicateMeta.podRequest
+		if predicateMeta.scalaResourcesHandleByExtenders != nil {
```

Comment:
Isn't `meta` an implementation of [this interface](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/algorithm/types.go#L165)? How are you accessing fields directly?

Reply:
nvm. I see we dynamically cast which isn't ideal though. @bsalamat FYI!